libs/langchain/langchain/vectorstores/timescalevector.py

```diff
@@ -0,0 +1,889 @@
+"""VectorStore wrapper around a Postgres-TimescaleVector database."""
+from __future__ import annotations
+
+import enum
+import logging
+import uuid
+from datetime import timedelta
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+)
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.utils import get_from_dict_or_env
+from langchain.vectorstores.base import VectorStore
+
+if TYPE_CHECKING:
+    from timescale_vector import Predicates
+
+
+class DistanceStrategy(str, enum.Enum):
+    """Enumerator of the Distance strategies."""
+
+    EUCLIDEAN = "l2"
+    COSINE = "cosine"
+    MAX_INNER_PRODUCT = "inner"
+
+
+DEFAULT_DISTANCE_STRATEGY = DistanceStrategy.COSINE
+
+ADA_TOKEN_COUNT = 1536
+
+_LANGCHAIN_DEFAULT_COLLECTION_NAME = "langchain_store"
+
+
+class TimescaleVector(VectorStore):
+    """VectorStore implementation using the timescale vector client to store vectores
+    in Postgres.
+
+    To use, you should have the ``timescale_vector`` python package installed.
+
+    Args:
+        service_url: Service url on timescale cloud.
+        embedding_function: Any embedding function implementing
+            `langchain.embeddings.base.Embeddings` interface.
+        collection_name: The name of the collection to use. (default: langchain_store)
+            This will become the table name used for the collection.
+        distance_strategy: The distance strategy to use. (default: COSINE)
+        pre_delete_collection: If True, will delete the collection if it exists.
+            (default: False). Useful for testing.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.vectorstores import TimescaleVector
+            from langchain.embeddings.openai import OpenAIEmbeddings
+
+            SERVICE_URL = "postgres://tsdbadmin:<password>@<id>.tsdb.cloud.timescale.com:<port>/tsdb?sslmode=require"
+            COLLECTION_NAME = "state_of_the_union_test"
+            embeddings = OpenAIEmbeddings()
+            vectorestore = TimescaleVector.from_documents(
+                embedding=embeddings,
+                documents=docs,
+                collection_name=COLLECTION_NAME,
+                service_url=SERVICE_URL,
+            )
+
+
+    """
+
+    def __init__(
+        self,
+        service_url: str,
+        embedding_function: Embeddings,
```

Comment:
can we call this `embedding` to match other vector stores

Reply:
Fixed in new commit